nikola/nikola.py

```diff
@@ -1986,38 +1986,27 @@ def scan_posts(self, really=False, ignore_quit=False, quiet=False):
             sys.exit(1)
         signal('scanned').send(self)
 
-    def generic_page_renderer(self, lang, post, filters, context=None):
-        """Render post fragments to final HTML pages."""
+    def generic_renderer(self, lang, output_name, template_name, filters, deps=[], uptodate_deps=[], pre_context=None, post_context=None, context_deps_remove=None, post_deps_dict=None, url_type=None):
+        """Helper function for rendering pages and post lists and other related pages."""
         utils.LocaleBorg().set_locale(lang)
-        context = context.copy() if context else {}
-        deps = post.deps(lang) + \
-            self.template_system.template_deps(post.template_name)
-        deps.extend(utils.get_asset_path(x, self.THEMES) for x in ('bundles', 'parent', 'engine'))
+
+        deps += self.template_system.template_deps(template_name)
         deps = list(filter(None, deps))
-        context['post'] = post
-        context['lang'] = lang
-        context['title'] = post.title(lang)
-        context['description'] = post.description(lang)
-        context['permalink'] = post.permalink(lang)
-        if 'pagekind' not in context:
-            context['pagekind'] = ['generic_page']
-        if post.use_in_feeds:
-            context['enable_comments'] = True
-        else:
-            context['enable_comments'] = self.config['COMMENTS_IN_STORIES']
-        extension = self.get_compiler(post.source_path).extension()
-        output_name = os.path.join(self.config['OUTPUT_FOLDER'],
-                                   post.destination_path(lang, extension))
+
+        context = copy(pre_context) if pre_context else {}
+        context["lang"] = lang
+        if post_context:
+            context.update(post_context)
+
         deps_dict = copy(context)
-        deps_dict.pop('post')
-        if post.prev_post:
-            deps_dict['PREV_LINK'] = [post.prev_post.permalink(lang)]
-        if post.next_post:
-            deps_dict['NEXT_LINK'] = [post.next_post.permalink(lang)]
+        if context_deps_remove:
```

Comment:
Do you mean what `context_deps_remove` does? That should now be covered in the argument documentation. Please tell me if that's enough.